src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -101,6 +102,56 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
     }
   }
 
+  private def declarationLookupFunction(decl: Declaration, inputs: Map[FullyQualifiedName, WdlValue]): String => WdlValue ={
+    def identifierLookup(string: String): WdlValue = {
+
+      /* This is a scope hierarchy to search for the variable `string`.  If `decl.scopeFqn` == "a.b.c"
+       * then `hierarchy` should be Seq("a.b.c", "a.b", "a")
+       */
+      val hierarchy = decl.scopeFqn.split("\\.").foldLeft(Seq.empty[String]) { (seq, part) =>
+        seq match {
+          case s:Seq[String] if s.nonEmpty => s :+ s"${seq.last}.$part"
+          case _ => Seq(s"$part")
+        }
+      }.reverse
+
+      /* Attempt to resolve the string in each scope */
+      val attemptedValues = hierarchy.map {scope => inputs.get(s"$scope.$string")}
+      attemptedValues.find(_.isDefined).getOrElse {
+        throw new WdlExpressionException(s"Could not find a value for $string")
+      }.get
+    }
+    identifierLookup
+  }
+
+  /* Some declarations need a value from the user and some have an expression attached to them.
+   * For the declarations that have an expression attached to it already, evaluate the expression
+   * and return the value for storage in the symbol store
+   */
+  def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs): Try[WorkflowCoercedInputs] = {
+    import scala.collection.mutable
+    val collected = mutable.Map[String, WdlValue]()
+    class NoFunctions extends WdlFunctions {
+      def getFunction(name: String): WdlFunction = throw new UnsupportedOperationException("No functions should be called in this test")
+    }
+
+    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
+
+    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
+      val value = decl.expression.get.evaluate(declarationLookupFunction(decl, collected.toMap ++ userInputs), new NoFunctions)
+      collected += (decl.fullyQualifiedName -> value.get)
+      decl.fullyQualifiedName -> value
+    }.toMap
+
+    val (successes, failures) = evaluatedDeclarations.partition {case (_, tryValue) => tryValue.isSuccess}
```

Comment:
We should start using validations & disjunctions soon, I don't think here is the place to start though